packages/tracing/src/transaction.ts

```diff
@@ -60,18 +67,22 @@ export class Transaction extends SpanClass implements TransactionInterface {
     return this._name;
   }
 
-  /** Setter for `name` property, which also sets `source` */
+  /** Setter for `name` property, which also sets `source` as custom */
   public set name(newName: string) {
-    this._name = newName;
-    this.metadata.source = 'custom';
+    this.setName(newName);
   }
```

Comment:
Q: Is there a reason we're maintaining this setter and setName? Always seemed a bit redundant to me.

Reply:
to be backwards compatible - we should maybe fix this for v8, will add a todo